phing/tasks/local-sync.xml

```diff
@@ -15,15 +15,21 @@
     </phingcall>
   </target>
 
-  <target name="local:sync" description="Synchronize local environment from remote (remote --> local)."
+  <!-- Sync each local multisite from remote aliases. -->
+  <target name="local:sync" description="Synchronize each local environment from remote (remote --> local).">
+    <yamlKeys file="${blt.config-files.project}" variable="multisite" outputProperty="multisite.names"/>
+    <foreach list="${multisite.names}" param="multisite.name" target="local:sync:site"/>
```

Comment:
Agreed, this is a bit heavy as I'm seeing it run with my own multisites and often wouldn't be necessary. A target like `local:sync:all` could work as a nice wrapper if you do indeed want to grab everything. 

I like the idea of separating out site config to a file for each site.